main.go

```diff
@@ -262,6 +269,57 @@ func createMessengerInProc(pk *ecdsa.PrivateKey, db client.Database) (*client.Me
 	return messenger, nil
 }
 
+
+func createMessengerWithDataSync(pk *ecdsa.PrivateKey, db client.Database) (*client.Messenger, error) {
+	// collect mail server request signals
+	signalsForwarder := newSignalForwarder()
+	go signalsForwarder.Start()
+
+	// setup signals handler
+	signal.SetDefaultNodeNotificationHandler(
+		filterMailTypesHandler(signalsForwarder.in),
+	)
+
+	nodeConfig, err := generateStatusNodeConfig(*dataDir, *fleet, *configFile)
+	if err != nil {
+		exitErr(errors.Wrap(err, "failed to generate node config"))
+	}
+
+	statusNode := node.New()
+
+	protocolGethService := gethservice.New(
+		statusNode,
+		&keysGetter{privateKey: pk},
+	)
+
+	services := []gethnode.ServiceConstructor{
+		func(ctx *gethnode.ServiceContext) (gethnode.Service, error) {
+			return protocolGethService, nil
+		},
+	}
+
+	if err := statusNode.Start(nodeConfig, services...); err != nil {
+		return nil, errors.Wrap(err, "failed to start node")
+	}
+
+	shhService, err := statusNode.WhisperService()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get Whisper service")
+	}
+
+	t := adapters.NewDataSyncWhisperTransport(shhService, pk)
```

Comment:
I would prefer to have a package called `datasync` or even `mvds` and then have `mvds.NewAdapter` here and `mvds.NewClient` later in the code.

Reply:
@mandrigin so I was thinking of this too, it would require a set of changes in adapters though to make certain things public that are currently not.

Reply:
Maybe, well, we can keep it as is then